sqlmesh/dbt/manifest.py

```diff
@@ -0,0 +1,224 @@
+        self._load_all()
+        return self._sources_per_package[package_name or self._project_name]
+
+    def macros(self, package_name: t.Optional[str] = None) -> MacroConfigs:
+        self._load_all()
+        return self._macros_per_package[package_name or self._project_name]
+
+    @property
+    def all_macros(self) -> t.Dict[str, t.Dict[str, MacroInfo]]:
+        self._load_all()
+        result: t.Dict[str, t.Dict[str, MacroInfo]] = defaultdict(dict)
+        for package_name, macro_configs in self._macros_per_package.items():
+            for macro_name, macro_config in macro_configs.items():
+                result[package_name][macro_name] = macro_config.info
+        return result
+
+    def _load_all(self) -> None:
+        if self._is_loaded:
+            return
+        self._load_models_and_seeds()
+        self._load_sources()
+        self._load_macros()
+        self._is_loaded = True
+
+    def _load_sources(self) -> None:
+        for source in self._manifest.sources.values():
+            source_dict = source.to_dict()
+            source_dict.pop("database", None)
+
+            source_config = SourceConfig(
+                **_config(source),
+                **source_dict,
+            )
+            self._sources_per_package[source.package_name][
+                source_config.source_name
+            ] = source_config
+
+    def _load_macros(self) -> None:
+        for macro in self._manifest.macros.values():
+            if macro.name.startswith("test_"):
+                continue
+
+            self._macros_per_package[macro.package_name][macro.name] = MacroConfig(
+                info=MacroInfo(
+                    definition=macro.macro_sql,
+                    depends_on=list(_macro_references(self._manifest, macro)),
+                ),
+                path=Path(macro.original_file_path),
+            )
+
+    def _load_models_and_seeds(self) -> None:
+        for node in self._manifest.nodes.values():
+            if node.resource_type not in ("model", "seed"):
+                continue
+
+            macro_references = _macro_references(self._manifest, node)
+
+            node_dict = node.to_dict()
+            node_dict.pop("database", None)
+            base_config = {**_config(node), **node_dict, "path": Path(node.original_file_path)}
+
+            if node.resource_type == "model":
+                self._models_per_package[node.package_name][node.name] = ModelConfig(
+                    sql=node.raw_code,
+                    dependencies=Dependencies(
+                        macros=macro_references,
+                        refs=_refs(node),
+                        sources=_sources(node),
+                    ),
+                    **base_config,
+                )
+            else:
+                self._seeds_per_package[node.package_name][node.name] = SeedConfig(
+                    dependencies=Dependencies(macros=macro_references),
```

Comment:
Do seeds need refs and sources for hooks?

Reply:
good point

Reply:
```
E           dbt.exceptions.ParsingError: Parsing Error
E
E             Seeds cannot depend on other nodes. dbt detected a seed with a pre- or post-hook
E             that calls 'ref', 'source', or 'metric', either directly or indirectly via other macros.
E
```